File: c++/string/constructor.cpp
#include <string>
#include <iostream>
#include <sstream>
#include <cctype>
#include <cassert>
using namespace std;

void constructor()
{
	string s1;
	string s2("Hello World");
	string s3(s2);
	string s4(4, 'C');
	string s5(s2.begin(), s2.begin()+5);
	string s6("Hello", 2);
	string s7(s2, 3);
	string s8(s2, 3, 2);
	
	assert(s1=="");
	assert(s2=="Hello World");
   	assert(s3==s2);
	assert(s4=="CCCC");
	assert(s5=="Hello");
	assert(s6=="He");
	assert(s7=="lo World");
	assert(s8=="lo");
}

int main(void)
{
	constructor();
	return 0;
}

File: java/string/Constructor.java
public class Constructor {
    public static void main(String[] args) {
        String s1 = new String("Hello World!");
        assert s1.equals("Hello World!");

        char [] helloArray = {'H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r', 'l', 'd', '!'};
        String s2 = new String(helloArray);
        assert s2.equals("Hello World!");

        String s3 = new String(helloArray, 2, 3);
        assert s3.equals("llo");

        String s4 = new String(s3);
        assert s4.equals(s3);
    }
}
